processing/src/main/java/io/druid/segment/filter/Filters.java

```
@@ -20,103 +20,62 @@
 package io.druid.segment.filter;
 
 import com.google.common.base.Function;
-import com.google.common.collect.Lists;
-import io.druid.query.filter.AndDimFilter;
-import io.druid.query.filter.BoundDimFilter;
+import com.google.common.base.Predicate;
+import com.google.common.collect.ImmutableList;
+import com.metamx.common.guava.FunctionalIterable;
 import io.druid.query.filter.DimFilter;
-import io.druid.query.filter.ExtractionDimFilter;
 import io.druid.query.filter.Filter;
-import io.druid.query.filter.InDimFilter;
-import io.druid.query.filter.JavaScriptDimFilter;
-import io.druid.query.filter.NotDimFilter;
-import io.druid.query.filter.OrDimFilter;
-import io.druid.query.filter.RegexDimFilter;
-import io.druid.query.filter.SearchQueryDimFilter;
-import io.druid.query.filter.SelectorDimFilter;
-import io.druid.query.filter.SpatialDimFilter;
 
-import javax.annotation.Nullable;
 import java.util.List;
 
 /**
  */
 public class Filters
 {
-  public static List<Filter> convertDimensionFilters(List<DimFilter> filters)
+  /**
+   * Convert a list of DimFilters (some of which may be null) to a list of Filters (none of which will be null).
+   * The resulting list may not be the same length.
+   *
+   * @param dimFilters list of DimFilters
+   *
+   * @return list of Filters
+   */
+  public static List<Filter> toFilters(List<DimFilter> dimFilters)
   {
-    return Lists.transform(
-        filters,
-        new Function<DimFilter, Filter>()
-        {
-          @Override
-          public Filter apply(@Nullable DimFilter input)
-          {
-            return convertDimensionFilters(input);
-          }
-        }
+    return ImmutableList.copyOf(
+        FunctionalIterable
+            .create(dimFilters)
+            .filter(
```

Should we throw an exception instead if a DimFilter was null?